sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/PermissionServiceImpl.java

```diff
@@ -70,14 +69,7 @@ public boolean isApplicationManager(
       SugoiUser sugoiUser, String realm, String userStorage, String application) {
     List<String> searchRoleList =
         getSearchRoleList(sugoiUser, realm, userStorage, application, applicationManagerRoleList);
-    return checkIfUserGetRoles(sugoiUser, searchRoleList);
-  }
-
-  @Override
-  public boolean isApplicationManager(SugoiUser sugoiUser, String realm) {
-    List<String> searchRoleList =
-        getSearchRoleList(sugoiUser, realm, "*", "*", applicationManagerRoleList);
-    return checkIfUserGetRoles(sugoiUser, searchRoleList);
+    return checkIfUserGetRoles(sugoiUser, searchRoleList) && isReader(sugoiUser, realm, null);
```

Comment:
I think you can be app manager without being reader on the realm

Reply:
If you're appmanager without being reader there is no problem on the api side but there is on ui. Indeed on the ui, there is an autocomplete field where user can search users and add them into his app's groups. If the user is just appmanager this autocomplete field will not work.